remove preQualification extension, update framework reference

duncandewhurst opened the pull request:
Closes #217

TO-DO:

- [x] Finalise II.2 (list of pre-qualified suppliers) guidance
- [x] Update [spreadsheet template](https://docs.google.com/spreadsheets/d/1sE3t1d1ZZI3El-VK8GxQgFhh3kueM9OwQxVYw5EMu7k/edit#gid=159397949)
- [x] Update examples
- [x] Update disclosure timing tables in section II
- ~~Update examples in the `ocds-show` repository, copy the files to the `ocds-show-ppp` repository, and then run `make update_examples`~~
- [ ] Update spreadsheet template and worked example URLs

duncandewhurst committed:
remove preQualification extension, update framework reference

jpmckinney commented:
There are a lot of conflicts with 1.0-dev (possibly because this PR also makes whitespace changes). You might want to restart the PR.

jpmckinney commented:
Changes ignoring whitespace: https://github.com/open-contracting-extensions/public-private-partnerships/pull/245/files?w=1

jpmckinney commented on docs/framework.md:
````diff
@@ -937,60 +937,30 @@ References to all signatories to the contract (including any financiers that are
-  
-```
+* In a single-stage procedure, the procuring entity invites suppliers to bid without submitting any prior information.
+* In a multi-stage procedure, the procuring entity issues an invitation to participate in the process based on which suppliers submit requests to participate. The procuring entity then assesses the submissions and establishes a list of qualified suppliers to invite to bid.
````
This paragraph is confusing.

```suggestion
* In a multi-stage procedure, suppliers submit requests to participate in a first stage. The procuring entity assesses these requests, and establishes a list of qualified suppliers to invite to bid. Qualified supplies then submit bids in a second stage. The procuring entity assesses the bids, and awards a contract to the winning bid.
```

jpmckinney commented on docs/framework.md:
```diff
@@ -999,16 +969,16 @@ P   U
 </div>
 
-The `preQualification/enquiryPeriod` field should be used to provide the period during which enquiries may be made and answered. 
+Use the `tender/period` field to provide the period during which the first stage is open for submissions, `period/endDate` should contain the closing date for submissions.
```
```suggestion
Use the `tender/period` field to provide the period during which the first stage is open for submissions. `period/endDate` should contain the closing date for submissions.
```

jpmckinney commented on docs/framework.md:
```diff
@@ -1151,130 +1121,32 @@ Information on the eligibility criteria for participants in the pre-qualificatio
-**Example**: See [section I.4](#i-4-project-economic-and-social-benefits) for JSON and flattened examples of the `documents` building block.
-
-### II.2. List of pre-qualified suppliers 
+In a single-stage procedure, the RFQ documents are embedded in the Request for Proposal documents.
```
What is the source of this statement? I wasn't aware that an RFQ could be part of an RFP; typically these are separate stages/processes.

duncandewhurst replied:
See the entry for Request For Proposal in https://ppp-certification.com/ppp-certification-guide/glossary

jpmckinney replied:
Aha. NIGP describes RFP and RFQ as separate sub-processes: https://www.nigp.org/resource/global-best-practices/Public-Private%20Partnership%20P3%20Facilities%20and%20Infrastructure%20Best%20Practice.pdf?dl=true

What APMG above is saying is that qualification/selection criteria are included in the RFP documents. I think it's clearer to say that, than to say "RFQ documents are embedded in the RFP documents".

jpmckinney replied:
Or, this statement from APMG for "One-stage tender" is clearer:

> A tender process in which the government issues its Request for Qualifications (RFQ) and Request for Proposal (RFP) together, and bidders submit their qualifications and bid as a single submission.

The original statement from AMPG for "Request for Proposal (RFP)" was:

> In one-stage open tender processes, the contents of a Request for Qualifications (RFQ) are embedded in the same set of documents, for example, the Request for Proposal (RFP) also includes the RFQ regulations.

This is somewhat different than the profile's text, because "documents embedded in documents" doesn't make sense, but "the contents of a Request for Qualifications (RFQ) are embedded", or "the Request for Proposal (RFP) also includes the RFQ regulations" does make sense.

duncandewhurst replied:
Good point. I updated the RFQ guidance in https://github.com/open-contracting-extensions/public-private-partnerships/pull/245/commits/475291c062d8b7ed4c3179258825bd557fba155e and added guidance for the RFP, which was missing from the revised guidance, in https://github.com/open-contracting-extensions/public-private-partnerships/pull/245/commits/7a95a3c70b5e7b40b5193b2e90cb962cd09842f6.

duncandewhurst committed:
Update docs/framework.md

Co-authored-by: James McKinney <26463+jpmckinney@users.noreply.github.com>

duncandewhurst committed:
Update docs/framework.md

Co-authored-by: James McKinney <26463+jpmckinney@users.noreply.github.com>

duncandewhurst committed:
update rfq guidance

duncandewhurst committed:
add RFP guidance

duncandewhurst committed:
update guidance for list of pre-qualified suppliers

duncandewhurst committed:
remove reference to qualification releaseTag code

duncandewhurst committed:
remove releaseTag codelist

jpmckinney commented:
> There are a lot of conflicts with 1.0-dev (possibly because this PR also makes whitespace changes). You might want to restart the PR.

Just checking that you're confident you can resolve all conflicts, before adding more commits?

duncandewhurst committed:
Merge branch '1.0-dev' into 217-remove-preQualification-extension, resolve merge conflicts

duncandewhurst commented:
> > There are a lot of conflicts with 1.0-dev (possibly because this PR also makes whitespace changes). You might want to restart the PR.
> 
> Just checking that you're confident you can resolve all conflicts, before adding more commits?

Thanks for the reminder, resolved in https://github.com/open-contracting-extensions/public-private-partnerships/pull/245/commits/0ba067b8164ce840f013abf71d901381b7767eb8

duncandewhurst committed:
update example

update pre-qualification modelling, generate from input template, correct identifiers, add missing fields, remove some repeated fields, correct timezones, use american spelling

duncandewhurst committed:
run build-profile.py, remove prequalification codes from tag enum

duncandewhurst committed:
remove qualification codelist csvs and tag enum

duncandewhurst committed:
Correct json pointers and field paths

duncandewhurst committed:
Update contract period guidance to match disclosure timing table.

duncandewhurst commented:
@jpmckinney this is now ready for review.

Given that I only remembered about the instructions in the readme after updating the examples directly in this repo, I thought it best for you to check you're happy with the updates in this PR, after which I'll go through the steps to update them in the `ocds-show` and `ocds-show-ppp` repositories.

Similarly, I've [updated a copy](https://docs.google.com/spreadsheets/d/1sE3t1d1ZZI3El-VK8GxQgFhh3kueM9OwQxVYw5EMu7k/edit#gid=263302309) of the spreadsheet template but I'll wait to update the URLs in the documentation until you've checked you're happy with the changes in this PR.

duncandewhurst committed:
update changelog

jpmckinney commented on docs/examples/full.json:
```diff
@@ -81,14 +81,14 @@
         {
           "id": "XX-GOV-99999999",
-          "name": "National Fibre Infrastructure",
+          "name": "National fiber Infrastructure",
```
Why? Capitalization is incorrect now.

duncandewhurst replied:
Whoops. Find and replace error in the course of switching to American spelling. Fixed in https://github.com/open-contracting-extensions/public-private-partnerships/pull/245/commits/5cfeab17c1996768ef7130bee4178189f885455d

jpmckinney commented on docs/framework.md:
```diff
@@ -939,52 +936,22 @@ References to all signatories to the contract (including any financiers that are
-P   U                 
-==  ==  ==  ==  ==  ==
+* In a single-stage procedure, the procuring entity invites suppliers to bid without submitting any prior information.
+* In a multi-stage procedure, suppliers submit requests to participate in a first stage. The procuring entity assesses these requests, and establishes a list of qualified suppliers to invite to bid. Qualified supplies then submit bids in a second stage. The procuring entity assesses the bids, and awards a contract to the winning bid.
```
```suggestion
* In a single-stage procedure, the procuring entity invites suppliers to submit bids, without submitting any prior information.
* In a multi-stage procedure, the procuring entity invites suppliers to submit requests to participate in a first stage (pre-qualification). The procuring entity assesses the requests, and establishes a list of qualified suppliers to invite to submit bids. Qualified suppliers then submit bids in a second stage. The procuring entity assesses the bids, and awards a contract to the winning bidder.
```

jpmckinney commented on docs/framework.md:
````diff
@@ -939,52 +936,22 @@ References to all signatories to the contract (including any financiers that are
-```
-
-**Flattened example** (showing top-level fields only)
+In OCDS, the `tender` section represents the invitation to participate and is also used to describe the procedure as a whole.
````
```suggestion
In OCDS, the `tender` section represents the invitation to participate (whether to submit bids, or to submit requests to participate) and is also used to describe the procedure as a whole.
```

jpmckinney commented on docs/framework.md:
```diff
@@ -939,52 +936,22 @@ References to all signatories to the contract (including any financiers that are
+> The act of testing prospective bidders to determine whether they meet the pass/fail qualification criteria in advance of issuing the request for proposals. ([Source](https://ppp-certification.com/ppp-certification-guide/glossary))
 
-</div>
+Use the `tender` section to describe the first stage of the contracting process, whether that is the tender itself, or a pre-qualification stage.
```
```suggestion
Use the `tender` section to describe the first stage of the contracting process, whether it involves the submission of bids, or requests to participate.
```

jpmckinney commented on docs/framework.md:
```diff
@@ -999,16 +966,15 @@ P   U
+Use the `tender/tenderPeriod` field to provide the period during which the first stage is open for submissions. `tenderPeriod/endDate` should contain the closing date for submissions.
 
-**Schema**: Information can be provided using a [Period](https://standard.open-contracting.org/1.1/en/schema/reference/#period) object in the [preQualification/enquiryPeriod](../reference/schema/#release-schema.json,,preQualification/enquiryPeriod) field.
+**Schema**: Information can be provided using a [Period](https://standard.open-contracting.org/1.1/en/schema/reference/#period) object in the [tender/period](../reference/schema/#release-schema.json,,tender/period) field.
```
```suggestion
**Schema**: Information can be provided using a [Period](https://standard.open-contracting.org/1.1/en/schema/reference/#period) object in the [tender/tenderPeriod](../reference/schema/#release-schema.json,,tender/tenderPeriod) field.
```

jpmckinney commented on docs/framework.md:
````diff
@@ -1151,124 +1116,26 @@ Information on the eligibility criteria for participants in the pre-qualificatio
-```eval_rst
-.. jsoninclude:: examples/full.json
-  :jsonpointer: /releases/1/tender/enquiryPeriod
+Request for Qualification (RFQ):
````
Do we need this line, which repeats the heading?

duncandewhurst replied:
It's in keeping with the pattern used in other headings: provide the name used in the framework under the OCDS for PPPs heading. Although, in this case, the two are identical. Happy to remove it if you prefer?

jpmckinney replied:
I suggest either removing it or putting it inside the following blockquote, so that it's clear it is being quoted from the framework.

duncandewhurst replied:
Removed in https://github.com/open-contracting-extensions/public-private-partnerships/pull/245/commits/ef0651de03805ab5f86c3493e16fb4452b0422ae

jpmckinney commented on docs/framework.md:
````diff
@@ -1458,55 +1276,39 @@ P   U
-```
-
-**JSON example:**
+Use [Document](reference/documents) objects in the [tender/documents](../reference/schema/#release-schema.json,,tender/documents) array to provide links to procurement documents, feasibility studies, including land acquisition, social, environmental, and rehabilitation related information and reports of independent procurement auditors. A short summary of each document can be provided using its ` ...
````
The "including land acquisition, social, environmental, and rehabilitation related information" in the middle of a list of document types reads a bit odd.

duncandewhurst replied:
Yes, it's based on this part of the framework which also reads oddly:

> links to all procurement documents, final feasibility study(including cost-benefit analysis, if any), land acquisition, rehabilitation, social (including information on human rights impact assessment) and environmental assessment, related information, and reports of independent procurement auditors (if any).

How about:

```markdown
Use [Document](reference/documents) objects in the [tender/documents](../reference/schema/#release-schema.json,,tender/documents) array to provide links to documents.

You should provide links to:

* all procurement documents 
* feasibility studies, including cost-benefit analyses, if available
* land acquisition, rehabilitation, social and environmental assessments, including information on human rights impact assessments
* reports of independent procurement auditors

A short summary of each document can be provided using its `description` field. Each document's `documentType` field should be set to a value from the [document type codelist](../reference/codelists/#documenttype), to identify the type of document being disclosed.
```

jpmckinney replied:
That's much clearer. I think we can change one bullet to:

```
* land acquisition, rehabilitation, social, human rights, and environmental assessments
```

The original wording I guess treats human rights assessments as a sub-type of social assessment, but I don't think this concept hierarchy is important to retain.

A clearer, easier to read, but longer format would be:

```
* land acquisition assessments
* rehabilitation assessments
* social assessments
* human rights assessments
* environmental assessments

duncandewhurst replied:
Updated in https://github.com/open-contracting-extensions/public-private-partnerships/pull/245/commits/ef0651de03805ab5f86c3493e16fb4452b0422ae

jpmckinney commented on docs/framework.md:
```diff
@@ -1458,55 +1276,39 @@ P   U
+### II.2. List of pre-qualified suppliers
 
-**Flattened example** (showing top-level fields only)
+Pre-qualification or shortlist.
```
Should this be:

```suggestion
> Pre-qualification or shortlist.
```

duncandewhurst replied:
I don't think so. The names from the framework provided under other headings aren't quoted. Only definitions from other sources are quoted. However, perhaps we could make it clearer that the names are taken directly from the framework.

jpmckinney replied:
I think it would be clearest for them all to be indicated somehow. I think quoting is fine for that purpose.

duncandewhurst replied:
In https://github.com/open-contracting-extensions/public-private-partnerships/pull/245/commits/ef0651de03805ab5f86c3493e16fb4452b0422ae I've quoted all definitions from the framework and removed any which simply repeat the headings.

jpmckinney replied:
I think maybe this specific one isn't quoted yet.

jpmckinney reviewed (commented):
Can you summarize the significant changes to the JSON files? They're hard to review.

jpmckinney commented:
> Given that I only remembered about the instructions in the readme after updating the examples directly in this repo, I thought it best for you to check you're happy with the updates in this PR, after which I'll go through the steps to update them in the `ocds-show` and `ocds-show-ppp` repositories.

That's fine - as long as we keep it all in sync to avoid overwriting changes in future.

duncandewhurst committed:
Fix capitalization of Fiber

duncandewhurst committed:
Update docs/framework.md

Co-authored-by: James McKinney <26463+jpmckinney@users.noreply.github.com>

duncandewhurst committed:
Update docs/framework.md

Co-authored-by: James McKinney <26463+jpmckinney@users.noreply.github.com>

duncandewhurst committed:
Update docs/framework.md

Co-authored-by: James McKinney <26463+jpmckinney@users.noreply.github.com>

duncandewhurst committed:
Update docs/framework.md

Co-authored-by: James McKinney <26463+jpmckinney@users.noreply.github.com>

duncandewhurst commented:
> Can you summarize the significant changes to the JSON files? They're hard to review.

I should've mentioned that I used OCDS Kit to generate `full_record_package.json` from `full.json` so only the latter requires a detailed review.

The main change is to adapt the example to reflect the changes to the framework reference.

Besides that, I fixed some issues:

* Use American spelling of 'Fiber'
* Correct identifiers which were incorrectly converted to numbers
* Correct inconsistent release dates
* Correct inconsistent timezones
* Reduce repetition between releases to make the example easier to maintain
* Use sequential identifiers rather than '123456'
* Add `name` to `unit`

In the course of the updates `parties/identifier/id` and `parties/identifier/scheme` were reordered for some reason.

duncandewhurst committed:
Merge branch '1.0-dev' into 217-remove-preQualification-extension

jpmckinney commented:
Thanks! `full.json` is a bit long to review, so I'll trust the changes based on your description.

duncandewhurst committed:
Update framework reference

Copy-edit description of II.1.10. Quote definitions from World Bank framework. Remove lines that repeated headings.

duncandewhurst committed:
Update docs/framework.md

Co-authored-by: James McKinney <26463+jpmckinney@users.noreply.github.com>

jpmckinney commented:
> Update spreadsheet template and worked example URLs

Does this need to be done in this PR, or separately?

> Update examples in the ocds-show repository, copy the files to the ocds-show-ppp repository, and then run make update_examples

I'm going to have another look at this. Maybe we can just delete them from those repositories.

jpmckinney commented:
> > Update examples in the ocds-show repository, copy the files to the ocds-show-ppp repository, and then run make update_examples
> 
> I'm going to have another look at this. Maybe we can just delete them from those repositories.

I've just deleted the examples from those other repositories, where they were unused.

The instruction about the examples was a hold-over from when ocds-show-ppp was entirely copied into this repository, which caused some real maintenance issues (changes here not being replicated to ocds-show or ocds-show-ppp). The examples themselves serve no purpose in those other repositories, so it's fine to edit them here only.

For reference: https://github.com/open-contracting-extensions/public-private-partnerships/pull/220

jpmckinney committed:
build: Remove instructions about where to edit examples
